product_code/matriculation_service/impl/src/test/scala/de/upb/cs/uc4/matriculation/impl/MatriculationServiceSpec.scala

```diff
@@ -305,6 +306,79 @@ class MatriculationServiceSpec extends AsyncWordSpec
       }
     }
 
+    "fail getting a proposal for adding matriculation data for a student, given a proposal while using another auth username" in {
+      val message = createSingleMatriculation(examReg0.name, "SS2020")
+      certificate.setup(student0.username)
+      certificate.getEnrollmentId(student0.username).invoke().flatMap { jsonId =>
+        client.getMatriculationProposal(student0.username).handleRequestHeader(addAuthorizationHeader(student0.username + "thisShouldFail"))
+          .invoke(message).failed.map { answer =>
+            answer.asInstanceOf[UC4Exception].possibleErrorResponse.`type` should ===(ErrorType.OwnerMismatch)
+          }.andThen {
+            case _ => cleanup()
+          }
+      }
+    }
```

Comment:
This test is fine for now, but when operations are merged, this test should be renamed to "[...] as another student", because admins are then allowed to get proposals for other students.

Reply:
```suggestion
    "fail getting a proposal for adding matriculation data for a student as another student" in {
      val message = createSingleMatriculation(examReg0.name, "SS2020")
      certificate.setup(student0.username)
      certificate.getEnrollmentId(student0.username).invoke().flatMap { jsonId =>
        client.getMatriculationProposal(student0.username).handleRequestHeader(addAuthorizationHeader(student0.username + "thisShouldFail"))
          .invoke(message).failed.map { answer =>
            answer.asInstanceOf[UC4Exception].possibleErrorResponse.`type` should ===(ErrorType.OwnerMismatch)
          }.andThen {
            case _ => cleanup()
          }
      }
    }
```